src/bootstrap.c

```diff
@@ -130,7 +137,7 @@ appling_bootstrap__on_signal(uv_async_t *handle) {
 }
 
 int
-appling_bootstrap(uv_loop_t *loop, js_platform_t *js, appling_bootstrap_t *req, const appling_key_t key, const char *dir, appling_bootstrap_cb cb) {
+appling_bootstrap(uv_loop_t *loop, js_platform_t *js, appling_bootstrap_t *req, const appling_key_t key, const char *dir, appling_bootstrap_cb cb, const char *link) {
```

Comment:
```suggestion
appling_bootstrap(uv_loop_t *loop, js_platform_t *js, appling_bootstrap_t *req, const appling_key_t key, const appling_link_t link, const char *dir, appling_bootstrap_cb cb) {
```

If the caller already has a parsed link, passing it to an API that requires a serialized link isn't trivial and requires heap allocation for the string. If on the other hand the caller has a serialized link, passing it to an API that requires a parsed link is as simple as doing an `appling_parse()` first which can be done without heap allocation.